internal/http/services/owncloud/ocdav/copy.go

```diff
@@ -52,52 +52,63 @@ func (s *svc) handlePathCopy(w http.ResponseWriter, r *http.Request, ns string)
 	ctx, span := rtrace.Provider.Tracer("reva").Start(r.Context(), "copy")
 	defer span.End()
 
-	src := path.Join(ns, r.URL.Path)
-	dst, err := extractDestination(r)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		return
-	}
-	for _, r := range nameRules {
-		if !r.Test(dst) {
+	// If is a third party copy
+	if r.Header.Get("ThirdPartyCopy") == "T" {
+
+		if r.Header.Get("Source") == "" { // Push Mode
+			s.handleTPCPush(ctx, w, r, ns)
+		} else { // Pull Mode
+			s.handleTPCPull(ctx, w, r, ns)
+		}
```

Comment:
@dynamic-entropy to avoid the else branch just `return` here

Reply:
Ah! Thankyou.